src/ReduceNoise/BM3DDenoise.jl

```diff
@@ -0,0 +1,65 @@
+using UUIDs
+const BM3DDenoise = Base.PkgId(UUID("95fb3b36-088a-43fb-bb1b-b1f34fadbd7d"), "BM3DDenoise")
+
+# Rewrite from ImageIO.jl
+function checked_import(pkgid)
+    Base.root_module_exists(pkgid) && return Base.root_module(pkgid)
+    # If not available, load the library or throw an error.
+    Base.require(pkgid)
+end
+
+
+@doc raw"""
+    BM3D(σ [, config=bm3d_config()])
+
+The BM3D(sparse 3D transform-domain collaborative filtering) denoising algorithm.
+
+!!! info
+
+    This feature depends on the package: `BM3DDenoise`
+
+    You can install it by `Pkg`:
+    ```
+    using Pkg
+    Pkg.add("BM3DDenoise")
+    ```
```

Comment:
Actually, I was thinking of directly add `BM3DDenoise` into `Project.toml` but not load it until required by users. This way we still benefit from the lazy loading.

The source code of `BM3DDenoise` is quite small so preinstall it for all users sounds good to me.

How do you think?

Reply:
I have no problem with that.

Does the existing loading mechanism need to be modified?
Modify `Project.toml` only?

Reply:
Yep, I think we only need to modify `Project.toml` and remove this "!!! info" section.